src/plugins/saved_objects_finder/public/finder/saved_object_finder.tsx

```diff
@@ -136,26 +116,22 @@ export class SavedObjectFinderUi extends React.Component<
         displayName: metadata.name,
       }))
     );
-    const hasReference = savedObjectsManagement.getTagFindReferences({
+    const includeTags = getTagFindReferences({
       selectedTags,
       taggingApi: this.props.services.savedObjectsTagging,
-    });
-    const params: FindQueryHTTP = {
-      type: visibleTypes ?? Object.keys(metaDataMap),
-      search: queryText ? `${queryText}*` : undefined,
-      fields: [...new Set(fields)],
-      page: 1,
-      perPage,
-      searchFields: ['title^3', 'description', ...additionalSearchFields],
-      defaultSearchOperator: 'AND',
-      hasReference: hasReference ? JSON.stringify(hasReference) : undefined,
-    };
+    })?.map(({ id, type }) => id);
 
-    const response = (await http.get('/internal/saved-objects-finder/find', {
-      query: params as Record<string, any>,
-    })) as FindResponseHTTP<FinderAttributes>;
+    const types = visibleTypes ?? Object.keys(metaDataMap);
+
+    const response = await contentClient.mSearch<SavedObjectCommon<FinderAttributes>>({
```

Comment:
My feeling is indeed that we should have our own interface here with the minimum set of fields that are consumed by apps (for ex is `coreMigrationVersion` or `typeMigrationVersion` or `namespaces` actually needed/consumed in the UI?)

Reply:
@jloleysens, I changed from `SavedObject` to `SOWithMetadata`, still far from ideal, but better as we more decoupled form saved objects. thanks 


@sebelga,
> My feeling is indeed that we should have our own interface here with the minimum set of fields that are consumed by apps

Agree, yes, if/when we move forward with cm this could be the direction. 
But for now `msearch `is just a different way to query saved objects with intermediate bwc layer. If to do this typing technically correct right now, then seems we can only type this as generic `object` for now :( 
Or we could do an intersection of all supported types and their interfaces